inferno-core/src/Inferno/Core.hs

```diff
@@ -53,19 +54,25 @@ data Interpreter m c = Interpreter
       Either InfernoError (Expr (Maybe VCObjectHash) SourcePos),
     parseAndInfer ::
       Text ->
-      Either InfernoError (Expr (Pinned VCObjectHash) SourcePos, TCScheme),
+      Either InfernoError (Expr (Pinned VCObjectHash) SourcePos, TCScheme, Map.Map (Location SourcePos) (TypeMetadata TCScheme), [Comment SourcePos]),
     -- | Evaluates all functions in given closure and creates a pinned env containing them
     mkEnvFromClosure ::
       Map.Map ExtIdent (Value c (ImplEnvM m c)) ->
       Map.Map VCObjectHash VCObject ->
       ImplEnvM m c (TermEnv VCObjectHash c (ImplEnvM m c)),
     -- | The default pinned env containing only the prelude
     defaultEnv ::
-      TermEnv VCObjectHash c (ImplEnvM m c)
+      TermEnv VCObjectHash c (ImplEnvM m c),
+    -- | The type of each name in this interpreter's prelude
+    nameToTypeMap ::
+      Map.Map (Maybe ModuleName, Namespace) (TypeMetadata TCScheme),
+    -- | The set of all type classes in this interpreter's prelude
+    typeClasses ::
+      Set.Set TypeClass
```

Comment:
Trying to share the computation of the type map and set of type classes in the LSP